lib/samples/create_buffers_around_points/create_buffers_around_points.dart

```diff
@@ -0,0 +1,396 @@
+            ),
+            Positioned(
+              top: 0,
+              left: 0,
+              right: 0,
+              child: Container(
+                padding: const EdgeInsets.all(8),
+                color: Colors.white.withOpacity(0.4),
+                child: Text(
+                  _status.label,
+                  textAlign: TextAlign.center,
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void _initializeSymbols() {
+    // Initialize the fill symbol for the buffer.
+    bufferFillSymbol
+      ..color = Colors.yellow.withOpacity(0.5)
+      ..outline = SimpleLineSymbol(
+        style: SimpleLineSymbolStyle.solid,
+        color: Colors.green,
+        width: 3,
+      );
+
+    // Initialize the tap point symbol.
+    tapPointSymbol
+      ..style = SimpleMarkerSymbolStyle.circle
+      ..color = Colors.red
+      ..size = 10;
+  }
+
+  // The build method for the Setting bottom sheet.
+  Widget buildSettings(BuildContext context, StateSetter setState) {
+    return Container(
+      padding: EdgeInsets.fromLTRB(
+        20.0,
+        0.0,
+        20.0,
+        max(
+          20.0,
+          View.of(context).viewPadding.bottom /
+              View.of(context).devicePixelRatio,
+        ),
+      ),
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Row(
+            children: [
+              Text(
+                'Settings',
+                style: Theme.of(context).textTheme.titleLarge,
+              ),
+              const Spacer(),
+              IconButton(
+                onPressed: () {
+                  Navigator.pop(context); // Close the bottom sheet
+                },
+                icon: const Icon(Icons.close),
+              ),
+            ],
+          ),
+          Row(
+            children: [
+              const Text('Buffer Radius (miles)'),
+              const Spacer(),
+              Text(
+                _bufferRadius.round().toString(),
+                textAlign: TextAlign.right,
+              ),
+            ],
+          ),
+          Row(
+            children: [
+              Expanded(
+                // A slider to adjust the buffer radius.
+                child: Slider(
+                  value: _bufferRadius,
+                  min: 10,
+                  max: 300,
+                  onChanged: (value) => setState(() => _bufferRadius = value),
+                ),
+              ),
+            ],
+          ),
+          Row(
+            children: [
+              Text(_shouldUnion ? 'Union Enabled' : 'Union Disabled'),
+              const Spacer(),
+              Switch(
+                value: _shouldUnion,
+                onChanged: (value) {
+                  setState(() {
+                    _shouldUnion = value;
+                    if (_bufferPoints.isNotEmpty) {
+                      drawBuffers(unionized: _shouldUnion);
+                    }
```

Comment:
this `if` should go outside the setState

Reply:
Ok . Updated code:
```dart
setState(() => _shouldUnion = value);
                  if (_bufferPoints.isNotEmpty) {
                    drawBuffers(unionized: _shouldUnion);
                  }```